tidy3d/web/webapi.py

```diff
@@ -702,14 +702,24 @@ def estimate_cost(task_id: str) -> float:
     task = SimulationTask.get(task_id)
     if not task:
         raise ValueError("Task not found.")
-    resp = task.estimate_cost()
-    flex_unit = resp.get("flex_unit")
-    if not flex_unit:
-        log.warning(
-            "Could not get estimated cost! It will be reported in preprocessing upon "
-            "simulation run."
-        )
-    return flex_unit
+
+    task.estimate_cost()
+    task_info = get_info(task_id)
+    status = task_info.metadataStatus
+
+    # Wait for a termination status
+    while status not in ["processed", "error"]:
+        time.sleep(REFRESH_TIME)
+        task_info = get_info(task_id)
+        status = task_info.metadataStatus
+
+    if status == "processed":
+        return task_info.estFlexUnit
+
+    log.warning(
+        "Could not get estimated cost! It will be reported in preprocessing upon simulation run."
```

Comment:
maybe confusing what is meant by "reported in preprocessing". Perhaps just say how to obtain it after the simulation run?

Reply:
This message was there even before btw. The estimate cost is printed during web.monitor, but only when the task status gets to "preprocess", because that's when we know the estimated flex unit should be populated and correct. That message also says that the user can call `web.real_cost` after the simulation run to get the billed cost.

How about "It will be reported during a simulation run in the preprocessing step."? Btw hopefully this message should never be encountered anymore.

Reply:
ok sounds good